Tests/SwiftParserTest/AttributeTests.swift

```diff
@@ -720,7 +720,42 @@ final class AttributeTests: ParserTestCase {
 
     assertParse(
       """
-      @attached(member, names: named(subscript(a:b:)))
+      @attached(declaration, names: named(subscript(a:b:)))
+      macro m()
+      """
+    )
+
+    assertParse(
+      """
+      @freestanding(declaration, names: named(deinit))
+      macro m()
+      """
+    )
+
+    assertParse(
+      """
+      @freestanding(declaration, names: named(init))
+      macro m()
+      """
+    )
+
+    assertParse(
+      """
+      @freestanding(declaration, names: named(init(a:b:)))
+      macro m()
+      """
+    )
+
+    assertParse(
+      """
+      @freestanding(member, names: named(subscript))
+      macro m()
+      """
+    )
+
+    assertParse(
+      """
+      @freestanding(member, names: named(subscript(a:b:)))
```

Comment:
I think it would make sense to add substructure tests here and check whether `subscript` etc. get parsed as a keyword or not?

Reply:
Alright